bazel/EXTERNAL_DEPS.md

```diff
@@ -1,4 +1,37 @@
-# Adding external dependencies to Envoy
+# Adding external dependencies to Envoy (native Bazel)
+
+This is the preferred style of adding dependencies that use Bazel for their
+build process.
+
+1. Specify name and version in [external documentation](../docs/install/requirements.rst).
+2. Define a new Bazel repository in [`bazel/repositories.bzl`](repositories.bzl),
+   in the `envoy_dependencies()` function.
+5. Reference your new external dependency in some `envoy_cc_library` via the
+   `external_deps` attribute.
+6. `bazel test //test/...`
+
+# Adding external dependencies to Envoy (genrule repository)
+
+This is the newer style of adding dependencies with no upstream Bazel configs.
+It wraps the dependency's native build tooling in a Bazel genrule, installing
+to a Bazel-managed prefix.
+
+1. Specify name and version in [external documentation](../docs/install/requirements.rst).
+2. Add a BUILD file in [`bazel/external/`](external/), using a genrule to build
+   the dependency. Please do not add BUILD logic that replaces the dependency's
+   upstream build tooling.
+2. Define a new Bazel repository in [`bazel/repositories.bzl`](repositories.bzl),
+   in the `envoy_dependencies()` function. The repository may use `genrule_repository`
+   from [`bazel/genrule_repository.bzl`](genrule_repository.bzl) to place large
+   genrule commands into a separate file.
+5. Reference your new external dependency in some `envoy_cc_library` via Y in the
+   `external_deps` attribute.
```

Comment:
Can you document how developers deal with external dependencies that depend on other external deps? It was a lot more direct before (we just has essentially a global prefix path).

Reply:
I'm not sure how to document this, because it'll be different for each dep -- some use Autotools and thus need flags to `./configure`, others use CMake and need ... whatever CMake does. I added a link to the Lightstep build command to use as an example.